### DISCUSSION TITLE
롬복 사용 관련 논의사항

### DISCUSSION THREAD

**Discussion by gkdud583** [prgrms-web-devcourse/Team-Saiko-BidMarket-BE#131] [Ideas]
1) @builder 어노테이션만으로도 문제 없는데 @RequiredArgsConstructor 붙이는 이유

2) @EqualsAndHashCode 사용시 발생할 수 있는 문제점
https://www.notion.so/backend-devcourse/c0954a42b659425e89be36728b62ace4?v=fba9be0e1028483fbc485c2f50704501&p=2dbee0d27b314529a5f8e302af4b5a3e&pm=s

-> 
1) 제 생각은 엔티티에서 불변인 값을 찾기가 어려우니 id기반으로 equals, hashcode 사용하되 영속화 전에 HashSet, HashMap 사용시 문제가 발생할수 있다는점을 인지하고 있어야 할 것 같습니다.
2) @EqualsAndHashCode 를 jpa buddy 가 추천하는 방식으로 변경하되 지연 로딩 되지 않는 필드만 사용해서 toString() 만드는식으로 하면 좋을 것 같습니다.

**Comment by UJ15** [comment]
꼭 읽고 인증 남겨주세요
